acceptance/acceptance_suite_test.go

```diff
@@ -0,0 +1,208 @@
+	"github.com/suse/carrier/cli/helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestAcceptance(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Acceptance Suite")
+}
+
+var nodeSuffix, nodeTmpDir string
+
+var _ = SynchronizedBeforeSuite(func() []byte {
+	if os.Getenv("REGISTRY_USERNAME") == "" || os.Getenv("REGISTRY_PASSWORD") == "" {
+		panic("REGISTRY_USERNAME and REGISTRY_PASSWORD environment variables can't be empty")
+	}
+
+	fmt.Printf("Compiling Carrier on node %d\n", config.GinkgoConfig.ParallelNode)
+
+	buildCarrier()
+	return []byte(strconv.Itoa(int(time.Now().Unix())))
+}, func(randomSuffix []byte) {
+	var err error
+
+	nodeSuffix = fmt.Sprintf("%d-%s",
+		config.GinkgoConfig.ParallelNode, string(randomSuffix))
+	nodeTmpDir, err = ioutil.TempDir("", "carrier-"+nodeSuffix)
+	if err != nil {
+		panic("Could not create temp dir: " + err.Error())
+	}
+
+	copyCarrier()
+	fmt.Printf("Creating a cluster for node %d\n", config.GinkgoConfig.ParallelNode)
+	createCluster()
+	os.Setenv("KUBECONFIG", nodeTmpDir+"/kubeconfig")
+
+	fmt.Printf("Creating image pull secret for Dockerhub on node %d\n", config.GinkgoConfig.ParallelNode)
+	helpers.Kubectl(fmt.Sprintf("create secret docker-registry regcred --docker-server=%s --docker-username=%s --docker-password=%s",
+		"https://index.docker.io/v1/",
+		os.Getenv("REGISTRY_USERNAME"),
+		os.Getenv("REGISTRY_PASSWORD"),
+	))
+
+	fmt.Printf("Installing Carrier on node %d\n", config.GinkgoConfig.ParallelNode)
+	installCarrier()
+
+	// Allow things to settle. Shouldn't be needed after we fix this:
+	// https://github.com/SUSE/carrier/issues/108
+	fmt.Printf("Waiting for things to settle on node %d\n", config.GinkgoConfig.ParallelNode)
+
+	WaitForInstallationToSettle(180)
+	fmt.Printf("Done waiting on node %d\n", config.GinkgoConfig.ParallelNode)
+})
+
+var _ = AfterSuite(func() {
+	fmt.Printf("Uninstall carrier on node %d\n", config.GinkgoConfig.ParallelNode)
+	out, _ := uninstallCarrier()
+	match, _ := regexp.MatchString(`Carrier uninstalled`, out)
+	if !match {
+		panic("Uninstalling carrier failed: " + out)
+	}
+	fmt.Printf("Deleting cluster on node %d\n", config.GinkgoConfig.ParallelNode)
+	deleteCluster()
+	fmt.Printf("Deleting tmpdir on node %d\n", config.GinkgoConfig.ParallelNode)
+	deleteTmpDir()
+})
+
+func createCluster() {
+	name := fmt.Sprintf("carrier-acceptance-%s", nodeSuffix)
+
+	if _, err := exec.LookPath("k3d"); err != nil {
+		panic("Couldn't find k3d in PATH: " + err.Error())
+	}
+
+	_, err := RunProc("k3d cluster create "+name, nodeTmpDir, false)
+	if err != nil {
+		panic("Creating k3d cluster failed: " + err.Error())
+	}
+
+	kubeconfig, err := RunProc("k3d kubeconfig get "+name, nodeTmpDir, false)
+	if err != nil {
+		panic("Getting kubeconfig failed: " + err.Error())
+	}
+	err = ioutil.WriteFile(path.Join(nodeTmpDir, "kubeconfig"), []byte(kubeconfig), 0644)
+	if err != nil {
+		panic("Writing kubeconfig failed: " + err.Error())
+	}
+}
```

Comment:
Future: Should wait here for the cluster to be properly present. I saw `carrier install` getting run while the cluster's `traefik` was still in creation. It ended with `carrier` installing its own traefik, and k3d never able to start its own.